File: packages/react-native-reanimated/src/css/svg/configs/svg/rect.ts
'use strict';
import type { RectProps } from 'react-native-svg';

import type { SvgStyleBuilderConfig } from './common';
import { commonSvgProps } from './common';

export const SVG_RECT_PROPERTIES_CONFIG: SvgStyleBuilderConfig<RectProps> = {
  ...commonSvgProps,
  x: true,
  y: true,
  rx: true,
  ry: true,
  width: true,
  height: true,
  opacity: true,
};

File: packages/react-native-reanimated/Common/cpp/reanimated/CSS/svg/configs/interpolators/rect.cpp
#include <reanimated/CSS/common/values/CSSKeyword.h>
#include <reanimated/CSS/common/values/CSSNumber.h>
#include <reanimated/CSS/configs/interpolators/utils.h>
#include <reanimated/CSS/interpolation/InterpolatorFactory.h>
#include <reanimated/CSS/svg/configs/interpolators/common.h>
#include <reanimated/CSS/svg/configs/interpolators/rect.h>
#include <reanimated/CSS/svg/values/SVGLength.h>

namespace reanimated::css {

const InterpolatorFactoriesRecord SVG_RECT_INTERPOLATORS = mergeInterpolators(
    SVG_COMMON_INTERPOLATORS,
    InterpolatorFactoriesRecord{
        {"x", value<SVGLength, CSSKeyword>(0)},
        {"y", value<SVGLength, CSSKeyword>(0)},
        {"width", value<SVGLength, CSSKeyword>(0)},
        {"height", value<SVGLength, CSSKeyword>(0)},
        {"rx", value<SVGLength, CSSKeyword>(0)},
        {"ry", value<SVGLength, CSSKeyword>(0)},
        {"opacity", value<CSSDouble>(1)},
    });

} // namespace reanimated::css
